Src/HALALMock/Services/ADC/ADC.cpp

```diff
@@ -77,38 +105,68 @@ void ADC::turn_on(uint8_t id){
 		ErrorHandler("LPTIM - %d - of ADC - %d - did not start correctly", timer.name, peripheral->handle);
 		return;
 	}
+*/ // Commented out because of the mock
+
 	peripheral->is_on = true;
 }
 
 float ADC::get_value(uint8_t id) {
 	Instance& instance = active_instances[id];
-	uint16_t raw = instance.peripheral->dma_data_buffer[instance.rank];
-	if(instance.peripheral->handle == &hadc3) {
-		return raw / MAX_12BIT * ADC_MAX_VOLTAGE;
+
+	EmulatedPin& emulated_pin = available_emulated_instances[instance];
+	if (emulated_pin.type != PinType::ADC) {
+		printf("Pin %s is not configured to be used for ADC usage. WARNING: it means that this pin \
+				has been reconfigured to be another peripheral WHILE this pin was already configured to be an ADC", emulated_pin);
+		return 0;
 	}
```

Comment:
I don't know about this if, the user shouldn't even try to change the pin type mid run, but if you want to keep it change the printf by an error handler